File: psets/pset1/mario-more/mario.c
#include <stdio.h>
#include <cs50.h>

int get_user_height(void);
void pyramid(int height);

int main(void)
{
    // Get user height input
    int height = get_user_height();

    pyramid(height);
}

int get_user_height(void)
{
    int n;

    // Prompt input for user height until get in range of 1 - 8
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8);

    return n;
}

void pyramid(int height)
{
    // Loop for the amount of height
    for (int i = 0; i < height; i++)
    {
        // Get the amount of hash needed
        int hash = i + 1;

        // The spaces is the height - hash
        int spaces = height - hash;

        // Do a loop for spaces first
        for (int s = 0; s < spaces; s++)
        {
            printf(" ");
        }

        // Do a loop for hash
        for (int h = 0; h < hash; h++)
        {
            printf("#");
        }

        // Print the 2 spaces for gap in current line
        for (int g = 0; g < 2; g++)
        {
            printf(" ");
        }

        // Print the hash again for the next pyramid
        for (int h = 0; h < hash; h++)
        {
            printf("#");
        }

        printf("\n");
    }

}

File: psets/pset6/sentimental-mario-more/mario.py
# Import the get int function from cs50
from cs50 import get_int

# Get the input from user in range of 1 to 8
while (True):
    height = get_int("Height: ")
    if(height in range(1, 9)):
        break

for i in range(height):
    # Get the amount of hash needed
    hash = i + 1

    # The spaces is the height - hash
    spaces = height - hash

    # Do a loop for spaces first
    for s in range(spaces):
        print(" ", end="")

    # Do a loop for hash
    for h in range(hash):
        print("#", end="")

    # Print the 2 spaces for gap in current line
    for g in range(2):
        print(" ", end="")

    # Print the hash again for the next pyramid
    for h in range(hash):
        print("#", end="")

    print("")